Fixed WithEndpoint to require a protocol, and be the same as the environment variable

Kaushal28 opened the pull request:
When using the OTLP exporter, the WithEndpoint option requires an hostname, without the protocol.
However, the X_ENDPOINT environment variables requires an endpoint, including the protocol.

This was misleading, as both config have very similar names, but expect different values.

So, Fixed WithEndpoint to require a protocol, and be the same as the environment variable

Kaushal28 committed:
Fixed WithEndpoint to require a protocol, and be the same as the environment variable

Kaushal28 committed:
added support for adding scheme if not already present

Kaushal28 committed:
Fixed unit tests

Fixed review comments

Fixed scheme addition

Revert "Fixed scheme addition"

This reverts commit 2c9cf15aaf8218d2d7575a3021cee48a37755629.

Revert "Revert "Fixed scheme addition""

This reverts commit 12e7600ef5372eff65e3fe4b9235e28b2e5cc864.

fixed string conca

Kaushal28 committed:
Did the same changes for otlpmetrics

shubham-bansal96 commented on CODEOWNERS:
```diff
@@ -12,6 +12,6 @@
 #
 
-* @jmacd @MrAlias @Aneurysm9 @evantorrie @XSAM @dashpole @MadVikingGod @pellared @hanyuancheung @dmathieu
+* @MrAlias @Aneurysm9 @evantorrie @XSAM @dashpole @MadVikingGod @pellared @hanyuancheung @dmathieu
```
Have you intentionally made changes in this file ? OR this file changed automatically using `make precommit` command ?

Kaushal28 replied:
Oh, I guess it's showing these changes because main branch is updated in upstream. Let me merge main into my current branch. Thanks for pointing out.

shubham-bansal96 commented on exporters/otlp/otlptrace/internal/otlpconfig/options_test.go:
```diff
@@ -92,10 +92,55 @@ func TestConfigs(t *testing.T) {
+		{
+			name: "Test With Endpoint without scheme",
+			opts: []otlpconfig.GenericOption{
+				otlpconfig.WithEndpoint("127.0.0.1:4318"),
```
I think we can assign this localhost address `127.0.0.1:4318` in a variable and then can use it accordingly.

Kaushal28 replied:
I could find that the same IP address is also being used in the same manner in other test files as well. So should we do this just for this file? @ApsTomar Please share your thoughts. Thanks.

ApsTomar replied:
If different IP addresses are being used at different places, then we can't amend other unit tests. We can only focus on our unit test. So, simply use a local variable to store this IP and use it here in your unit test. Also, take into consideration the scope of that variable so that you don't have to declare it again and again in every test case.

Kaushal28 replied:
So I checked the unit tests and it seems that there are different ports and schemes being used with this IP to test different scenarios. So if I store only `127.0.0.1` in a variable, it'll increase the unnecessary sting formatting. Also, if we declare this variable at package level and only use in our test case, it might not be consistent as all other test cases do not use it. @shubham-bansal96 , @ApsTomar if you guys still think we should make this change, please let me know. Thanks.

opentelemetrybot committed:
dependabot updates Sun May  7 14:59:33 UTC 2023 (#4072)

Bump google.golang.org/grpc from 1.54.0 to 1.55.0 in /bridge/opentracing/test
Bump github.com/prometheus/client_golang from 1.15.0 to 1.15.1 in /example/view
Bump github.com/prometheus/client_model from 0.3.0 to 0.4.0 in /exporters/prometheus
Bump github.com/prometheus/client_golang from 1.15.0 to 1.15.1 in /exporters/prometheus
Bump google.golang.org/grpc from 1.54.0 to 1.55.0 in /exporters/otlp/otlptrace
Bump google.golang.org/grpc from 1.54.0 to 1.55.0 in /exporters/otlp/otlpmetric/otlpmetricgrpc
Bump google.golang.org/grpc from 1.54.0 to 1.55.0 in /exporters/otlp/otlptrace/otlptracegrpc
Bump google.golang.org/grpc from 1.54.0 to 1.55.0 in /exporters/otlp/otlpmetric
Bump google.golang.org/grpc from 1.54.0 to 1.55.0 in /example/otel-collector
Bump github.com/prometheus/client_golang from 1.15.0 to 1.15.1 in /example/prometheus
Bump golang.org/x/sys from 0.7.0 to 0.8.0 in /sdk

MrAlias committed:
Remove the deprecated `metric/instrument` package (#4055)

* Remove deprecated `metric/instrument`

* Apply suggestions from code review

---------

Co-authored-by: Robert Pająk <pellared@hotmail.com>

MrAlias committed:
Update metric iface warnings (#4056)

Remove self-reference to the package in the expandable interfaces.

MrAlias committed:
Add comment on all uses of embedded types (#4059)

MrAlias committed:
Fix the broken [Option] link in metric docs (#4057)

The `Option` type was replaced with `InstrumentOption` when the
`metric/instrument` package was moved to `metric`.

tsloughter committed:
Add semconv/v1.19.0 (#3848)

* Add semconv/v1.19.0

Includes adding semconv/internal/v3/ to support the move of
the semantic convention HTTPUserAgentKey to the new key
UserAgentOriginalKey

* Update CHANGELOG.md

Co-authored-by: Robert Pająk <pellared@hotmail.com>

* format http.go.tmpl using tabs instead of spaces

* update http.go.tmpl to use v3 of semconv/internal

* Generate attribute_group.go

---------

Co-authored-by: Robert Pająk <pellared@hotmail.com>
Co-authored-by: Anthony Mirabella <a9@aneurysm9.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

MrAlias committed:
Fix article in metric API docs (#4054)

MrAlias committed:
Document how to pass attributes for all record methods (#4058)

* Document how to pass attributes for all record methods

* Update based on feedback

tijmenstor committed:
feat(gh-workflows): Use cache-dependency-path in actions/setup-go for ci workflow (#4074)

Using cache-dependency-path (which is now a built-in functionality in actions/setup-go), the speed of runs on MacOS can be improved.

Refs: #3911

Co-authored-by: Robert Pająk <pellared@hotmail.com>
Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

ChillOrb committed:
added version.go and test file for issue 2143 (#3677)

* added version.go and test file for issue 2143

* added license in metric versiont_test

* goimport file linting

* update trace version to v.1.13.0 and metric to 0.36.0

* Update sdk/metric/version.go

Co-authored-by: Robert Pająk <pellared@hotmail.com>

* using asser.Regxp

* changing regex string as per recommendations

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>

* changing regex string as per recommendations

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>

* reverting go mod and go sum changes

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>

* trace and metric version bump up

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>

* version update in sdk/metric , sdk/trace

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>

* doc typo fix

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>

* Apply suggestions from code review

---------

Signed-off-by: ChillOrb <rakshitparashar1@gmail.com>
Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Robert Pająk <pellared@hotmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

pellared committed:
[chore] Pin otel-specification hyperlinks (#4033)

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

MrAlias committed:
Add measurement docs to metric API (#4053)

* Add measurement docs to metric API

Based on user feedback, how measurements are made with observable
instruments is not immediately clear to new users. This adds
documentation intended to help guide these users.

* Apply feedback

* Apply feedback

* Remove backticks

* Apply suggestions from code review

Co-authored-by: Robert Pająk <pellared@hotmail.com>

* Update metric/meter.go

Co-authored-by: Robert Pająk <pellared@hotmail.com>

---------

Co-authored-by: Robert Pająk <pellared@hotmail.com>
Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>

pellared committed:
sdk/resource: Fix build for BSD OSes (#4077)

* sdk/resource: Fix build for BSD OSes

* Add changelog

* Update header in changelog

---------

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

pellared committed:
sdk/resource: Add Telemetry Stability label (#4080)

* Add Telemetry Stability label

* Update sdk/resource/doc.go

* Update doc.go

---------

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>

dependabot[bot] committed:
Bump golang.org/x/tools from 0.8.0 to 0.9.1 in /internal/tools (#4087)

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.8.0 to 0.9.1.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.8.0...v0.9.1)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

opentelemetrybot committed:
dependabot updates Sun May 14 14:40:44 UTC 2023 (#4089)

Bump github.com/cloudflare/circl from 1.3.1 to 1.3.3 in /internal/tools

chalin committed:
[website_docs] Update path to spec, run formatter (#4084)

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>

ChenX1993 committed:
[SDK][Trace] Add tests for default id generator (#4043)

Signed-off-by: Chen Xu <chen.x@uber.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

pellared committed:
Add semconv/v1.20.0 (#4078)

* Add semconv/v1.20.0

* Update changelog

* Change http.flavor to net.protocol.(name|version)

* Update comments in httpconv

* Fix vanity import

* Update CHANGELOG.md

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

---------

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

ash2k committed:
Use strings.Cut() instead of string.SplitN() (#4049)

strings.Cut() generates less garbage as it does not allocate the slice to hold parts.

pellared committed:
[chore] Run generate before lint (#4093)

CharlieTLe committed:
feat(ci): Add codespell to Makefile and GitHub Actions (#3996)

* feat(ci): Add codespell to Makefile and GitHub Actions

This is a followup from
https://github.com/open-telemetry/opentelemetry-go/pull/3980 to add the
automation for checking for code spelling errors to a different PR.

[Codespell][] makes use of the `.codespellrc` file automatically as a config
file for convenience. The configuration file specifies directories and
flags to pass to `codespell`. Codespell also makes use of the
`.codespellignore` file to ignore certain word spellings.

There is a new section in `Makefile` for installing python tooling.
It is similar to how the go tools are installed. It sets up a virtualenv
in `venv `and installs python tooling there if it hasn't been installed
yet.

The new [codespell workflow][] will run in pull requests to annotate
misspelled words. It is set up to only warn the user with annotations in
the pull request if there are typos, but we can fail the pull request as
well if we want to by deleting the [warn_only][] key.

[codespell]: https://github.com/codespell-project/codespell
[codespell workflow]: https://github.com/codespell-project/actions-codespell
[warn_only]: https://github.com/codespell-project/actions-codespell#parameter-only_warn

* Use docker to run python and codespell

Since this is a Go project, having a dependency on python seemed
disconcerting. So, docker is used to create a virtual environment
with a python image and also run the tools that were installed.

* Remove wording on requirement for python

* Apply suggestions from code review

Co-authored-by: Damien Mathieu <42@dmathieu.com>

* Pin versions into requirements.txt

* Pin version in requirements.txt

* Fix typo in license

* Revert typo fix on deleted file

* Update .github/workflows/codespell.yaml

Co-authored-by: Robert Pająk <pellared@hotmail.com>

* Update codespell.yaml

* Update codespell.yaml

---------

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Damien Mathieu <42@dmathieu.com>
Co-authored-by: Robert Pająk <pellared@hotmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

andrew-womeldorf committed:
sdk/metric: Fix import comments (#4086)

* Resolve inconsistent import comments in sdk/metric

* update changelog

* Apply suggestions from code review

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

* Update CHANGELOG.md

---------

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>
Co-authored-by: Robert Pająk <pellared@hotmail.com>

dependabot[bot] committed:
Bump lycheeverse/lychee-action from 1.7.0 to 1.8.0 (#4104)

Bumps [lycheeverse/lychee-action](https://github.com/lycheeverse/lychee-action) from 1.7.0 to 1.8.0.
- [Release notes](https://github.com/lycheeverse/lychee-action/releases)
- [Commits](https://github.com/lycheeverse/lychee-action/compare/v1.7.0...v1.8.0)

---
updated-dependencies:
- dependency-name: lycheeverse/lychee-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

opentelemetrybot committed:
dependabot updates Sun May 21 15:20:04 UTC 2023 (#4126)

Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/zipkin
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /bridge/opentracing/test
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /schema
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/internal/retry
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/prometheus
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/stdout/stdoutmetric
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/otlptrace
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/otlpmetric
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/otlpmetric/otlpmetricgrpc
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/otlpmetric/otlpmetrichttp
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/otlptrace/otlptracegrpc
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/stdout/stdouttrace
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/otlp/otlptrace/otlptracehttp
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /metric
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /trace
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /exporters/jaeger
Bump codecov/codecov-action from 3.1.3 to 3.1.4
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /sdk/metric
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /bridge/opencensus
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /bridge/opentracing
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3 in /sdk
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3

MrAlias committed:
Release v1.16.0/v0.39.0 -- Stable Metric API (#4100)

* Update versions

* Prepare stable-v1 for version v1.16.0

* Prepare experimental-metrics for version v0.39.0

* Update changelog

pellared committed:
[chore] Refine generate Make target (#4097)

* [chore] Refine generate Make target

* Do not run porto in go-generate

* Update Makefile

---------

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

serdarkalayci committed:
Marked Metrics as stable (#4129)

pellared committed:
[chore] Update documentation guidelines (#4136)

* Simplify documentation guidelines

* Add info how to localy run pkg.go.dev

Aneurysm9 committed:
Add @pellared as maintainer (#4138)

* Add @pellared as maintainer

Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>

* Add @pellared as maintainer in CODEOWNERS

---------

Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>
Co-authored-by: Tyler Yahn <codingalias@gmail.com>

Aneurysm9 committed:
Move @Aneurysm9 to approver (#4137)

* Move @Aneurysm9 to approver

Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>

* Remove @Aneurysm9 from CODEOWNERS

---------

Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>
Co-authored-by: Tyler Yahn <codingalias@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

pellared committed:
[chore] Clarify that internal and test packages do not need docs (#4140)

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

dependabot[bot] committed:
Bump codecov/codecov-action from 3.1.3 to 3.1.4 (#4108)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 3.1.3 to 3.1.4.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v3.1.3...v3.1.4)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pellared committed:
sdk/metrics: Fix MeterProvider.Meter Go doc (#4152)

pellared committed:
sdk/metric: Log empty meter name (#4153)

* sdk/metric: Log empty meter name

* Avoid magic numbers

* Lint

* Apply suggestions from code review

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

* Update log message

* Apply suggestion

---------

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

pellared committed:
[chore] Update Project Status and README formatting (#4146)

* [chore] Update Project Status

* Fornat table

* Update README.md

Co-authored-by: Damien Mathieu <42@dmathieu.com>

---------

Co-authored-by: Damien Mathieu <42@dmathieu.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/golangci/golangci-lint from 1.52.2 to 1.53.2 in /internal/tools (#4177)

* Bump github.com/golangci/golangci-lint in /internal/tools

Bumps [github.com/golangci/golangci-lint](https://github.com/golangci/golangci-lint) from 1.52.2 to 1.53.2.
- [Release notes](https://github.com/golangci/golangci-lint/releases)
- [Changelog](https://github.com/golangci/golangci-lint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/golangci/golangci-lint/compare/v1.52.2...v1.53.2)

---
updated-dependencies:
- dependency-name: github.com/golangci/golangci-lint
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update golangci lint configuration

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tyler Yahn <codingalias@gmail.com>

pellared committed:
semconv: Stop generating httpconv.go, netconv.go, http.go (#4145)

dmathieu committed:
Return noop meters once the provider has been shutdown (#4154)

pellared committed:
[chore] Update dependabot-pr branch name (#4191)

opentelemetrybot committed:
dependabot updates Mon Jun  5 09:32:52 UTC 2023 (#4198)

Bump go.opentelemetry.io/build-tools/crosslink from 0.7.0 to 0.8.0 in /internal/tools
Bump go.opentelemetry.io/build-tools/dbotconf from 0.7.0 to 0.8.0 in /internal/tools
Bump go.opentelemetry.io/build-tools/multimod from 0.7.0 to 0.8.0 in /internal/tools
Bump go.opentelemetry.io/build-tools/semconvgen from 0.7.0 to 0.8.0 in /internal/tools
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/otlptrace
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/zipkin
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/otlptrace/otlptracehttp
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /bridge/opentracing/test
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/internal/retry
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/otlpmetric/otlpmetrichttp
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/otlptrace/otlptracegrpc
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/jaeger
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/stdout/stdouttrace
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/stdout/stdoutmetric
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/otlpmetric
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /schema
Bump github.com/itchyny/gojq from 0.12.12 to 0.12.13 in /internal/tools
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /metric
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /trace
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/prometheus
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /exporters/otlp/otlpmetric/otlpmetricgrpc
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /bridge/opencensus
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /sdk/metric
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /bridge/opentracing
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4 in /sdk

MrAlias committed:
Warn instead of Info log instrument conflict (#4202)

* Warn instead of Info log instrument conflict

* Add change to changelog

svrnm committed:
Common weights for go pages (#4211)

MrAlias committed:
Upgrade go.opentelemetry.io/proto/otlp to v0.20.0 (#4213)

pellared committed:
[chore] Add go-work Make target (#4199)

Co-authored-by: Chester Cheung <cheung.zhy.csu@gmail.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

parauliya committed:
Merge pull request #7 from open-telemetry/main

Update the main branch of the fork

Commit:
Update the review comment on PR 4160
